src/NumericalAlgorithms/Spectral/Python/Mesh.cpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <string>
+
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+namespace py = pybind11;
+
+namespace py_bindings {
+
+void bind_basis(py::module& m) {  // NOLINT
+
+  py::enum_<Spectral::Basis>(m, "Basis")
+      .value("Legendre", Spectral::Basis::Legendre)
+      .value("Chebyshev", Spectral::Basis::Chebyshev)
+      .value("FiniteDifference", Spectral::Basis::FiniteDifference);
+}
+
+void bind_quadrature(py::module& m) {  // NOLINT
+
+  py::enum_<Spectral::Quadrature>(m, "Quadrature")
+      .value("Gauss", Spectral::Quadrature::Gauss)
+      .value("GaussLobatto", Spectral::Quadrature::GaussLobatto)
+      .value("CellCentered", Spectral::Quadrature::CellCentered)
+      .value("FaceCentered", Spectral::Quadrature::FaceCentered);
+}
+
+template <size_t Dim>
+void bind_mesh(py::module& m) {  // NOLINT
+  //the bindings here are not complete
+  py::class_<Mesh<Dim>>(m, ("Mesh" + std::to_string(Dim) + "D").c_str())
```

Comment:
I think it would be useful to add a property `dim` to the Python class which is 1, 2 or 3 and static if possible. Would that be easy enough to add?